frontend/lib/util/filesystem.cpp

```diff
@@ -293,6 +352,49 @@ std::string cleanLocalPath(std::string path) {
   return path;
 }
 
+static bool filePathInDirPath(const char* filePathPtr, size_t filePathLen,
+                              const char* dirPathPtr, size_t dirPathLen) {
+  if (dirPathLen == 0)
+    return false; // documented behavior; use "." for the current dir.
+
+  // create SmallVectors for the relevant paths so we can use LLVM Path stuff
+  auto path = llvm::SmallVector<char>(filePathPtr, filePathPtr+filePathLen);
+  auto dirPath = llvm::SmallVector<char>(dirPathPtr, dirPathPtr+dirPathLen);
+
+  // set 'path' to filePath without the filename (i.e. the directory)
+  auto style = llvm::sys::path::Style::posix;
+  remove_filename(path, style);
+  remove_dots(path, /* remove_dot_dot */ false, style);
+  // remove_dots on foo.chpl returns "" but we want to match "." in that case
+  if (path.size() == 0)
+    path.push_back('.');
+
+  // also normalize dirPath
+  remove_dots(dirPath, /* remove_dot_dot */ false, style);
+  if (dirPath.size() == 0)
+    dirPath.push_back('.');
+
+  // add / to the end of path and dirPath if they are not present already
+  if (path.back() != '/')
+    path.push_back('/');
+  if (dirPath.back() != '/')
+    dirPath.push_back('/');
+
+  // now, check that 'dirPath' is a prefix or equal to 'path'
+  return dirPath.size() <= path.size() &&
```

Comment:
Should we use `llvm::sys::path::starts_with`?

Reply:
It'd be nice to add a TODO comment but I am short on time at the moment and so I'm going to merge this without it.